docs/messaging/messaging-metrics.md

```diff
@@ -93,92 +92,76 @@ the broker doesn't have such notion, the destination name SHOULD uniquely identi
 <!-- endsemconv -->
 
-### Metric: `messaging.publish.messages`
+## Producer metrics
+
+### Metric: `messaging.client.published.messages`
 
 This metric is [required][MetricRequired] when the messaging system supports batch publishing. It's [opt-in][MetricOptIn] when the messaging system does not support batch publishing, since the message count can be derived from the `messaging.publish.duration` histogram.
 
-<!-- semconv metric.messaging.publish.messages(metric_table) -->
+<!-- semconv metric.messaging.client.published.messages(metric_table) -->
 | Name     | Instrument Type | Unit (UCUM) | Description    | Stability |
 | -------- | --------------- | ----------- | -------------- | --------- |
-| `messaging.publish.messages` | Counter | `{message}` | Measures the number of published messages. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
+| `messaging.client.published.messages` | Counter | `{message}` | Measures the number of published messages. | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
 <!-- endsemconv -->
 
 > The need to report `messaging.publish.messages` depends on the messaging system capabilities and not application scenarios or client library limitations. For example, RabbitMQ does not support batch publishing and corresponding instrumentations don't need to report `messaging.publish.messages`. Kafka supports both, single and batch publishing, and instrumentations MUST report `messaging.publish.messages` counter regardless of application scenarios or APIs available in the client library.
 
 ## Consumer metrics
 
-### Metric: `messaging.receive.duration`
+### Metric: `messaging.client.consumed.messages`
 
-This metric is [required][MetricRequired] for operations that are initiated by the application code (pull-based).
+This metric measures number of messages that were consumed by the application. If messaging client supports pull mode, this metric SHOULD represent the number of successfully pulled
+messages. If messaging client pushes messages to the application, it SHOULD represent the number of messages that were dispatched to the application.
 
-This metric SHOULD be specified with
-[`ExplicitBucketBoundaries`](https://github.com/open-telemetry/opentelemetry-specification/tree/v1.31.0/specification/metrics/api.md#instrument-advice)
-of `[ 0.005, 0.01, 0.025, 0.05, 0.075, 0.1, 0.25, 0.5, 0.75, 1, 2.5, 5, 7.5, 10 ]`.
+The message SHOULD be counted each time it's received and consumed.
 
-When this metric is reported alongside a messaging receive span, the metric value SHOULD be the same as the corresponding span duration.
+If instrumentation tracks pull-receiving and processing for one message, it SHOULD track this message once within `messaging.client.consumed.messages` metric.
```

Comment:
this is probably not right. Knowing if messages are dropped between receiving and processing would be really useful, so maybe we need to count both operations.